resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -168,24 +168,34 @@ private[spark] class Client(
     val initContainerConfigMap = initContainerComponentsProvider
```

Comment:
And we shouldn't be making a config map if we're not using an init-container. In general we should try to encapsulate everything related to whether or not we need an init-container in one place, so that we only have to make the check once and run everything that relates to that check there.